Source/Csla/DataPortalClient/DataPortalProxy.cs

```diff
@@ -76,7 +76,7 @@ public async virtual Task<DataPortalResult> Create(Type objectType, object crite
         request.CriteriaData = SerializationFormatterFactory.GetFormatter(ApplicationContext).Serialize(criteria);
         request = ConvertRequest(request);
         var serialized = SerializationFormatterFactory.GetFormatter(ApplicationContext).Serialize(request);
-        serialized = await CallDataPortalServer(serialized, "create", GetRoutingToken(objectType), isSync);
+        serialized = await CallDataPortalServer(serialized, "create", GetRoutingToken(objectType), isSync).ConfigureAwait(false);
```

Comment:
I'm not sure this will work with Windows Forms. Maybe iffy in WPF too.

The thing is, the proxy gets called by the UI, and the results need to be marshalled onto the UI thread, because they are often directly bound to the UI upon return.

If this works in WinForms/WPF without breaking data binding, then I'm fine with it - but that needs to be explicitly tested.

Reply:
I'll test this to make sure. But imho this can't be in any data binding. We are still handling a `byte[]` here. So there is no real BO which could be data binded.

Reply:
I just remember having a lot of trouble getting the HttpProxy working reliably with WinForms and WPF and data binding.

If nothing between the async call and the UI refreshing data binding marshals the results back onto the UI thread/context, the code can remain on the wrong thread/context.

Reply:
Mhm okay. The context is captured on each await so ultimately as long as the user does not `IDataPortal.FetchAsync().ConfigureAwait(false);` it should be coming back on the correct context.
But yeah. I'll test that again.

Reply:
The issue might be that the sync APIs delegate through the async path? I don't think that's the case, but we should make sure that sync calls don't break in WinForms either.

Reply:
Do we have sample project for winforms?
I'm currently testing this change with the app from my workplace and so far no issues there.

Reply:
Yes, there's the simple one

https://github.com/MarimerLLC/csla/tree/main/Samples/WinFormsExample

And the one that is perhaps more important for this particular test:

https://github.com/MarimerLLC/csla/tree/main/Samples/RootChildGrandchildWinFormTest

Reply:
@StefanOssendorf please let me know when you have tested and ensured it doesn't cause issues.

Reply:
Yeah I'm at it. Locally these changes have no impact at all. Because the local proxy overrides these methods. And with an actual http call have I have no issues with our wpf App to navigate and work through our views

Reply:
@rockfordlhotka I tested this with the rootchildgrandchildwinforms sample and I can't spot a difference from the main version without my changes. And as described in our WPF app it's working fine.